calculator/calculator.py

```diff
@@ -0,0 +1,218 @@
+    """Check the existence of the operands.
+
+    Args:
+        func (function): the function to operate on.
+
+    Returns:
+        function: the function wrapper.
+    """
+
+    def operands_wrapper(operator_symbol, operands):
+        left_operand, right_operand = operands
+
+        if left_operand and right_operand:
+            return func(operator_symbol, operands)
+
+        elif not left_operand and right_operand:
+            # if any([operator_symbol == "+", operator_symbol == "-"]):
+            #     return float(f"{operator_symbol}{right_operand}")
+            return func(operator_symbol, [right_operand])
+
+        elif not right_operand and left_operand:
+            return func(operator_symbol, [left_operand])
+
+    return operands_wrapper
+
+
+@check_operands
+def calculate_sub_expression(symbol: str, operands: list) -> float:
+    """Evaluate an expression.
+
+    Args:
+        operands (list): the operands, can be 2 or 1 according to operation.
+        symbol (str): a symbol represents the mathematical operation.
+
+    Returns:
+        A float, the result of the mathematical operation.
+
+    Note:
+        Brackets need a string operand (_evaluate() takes a string) and the
+        others a float (factorial and negate).
+    """
+    if len(operands) == 2:
+        return operators[symbol].operation(float(operands[0]),
+                                           float(operands[1]))
+
+    operand = operands[0]
+    operation_to_do = operators[symbol].operation
+
+    if operation_to_do is not _evaluate:
+        operand = float(operands[0])
+
+    return operation_to_do(operand)
+
+
+def parse_sub_expression(symbol: str, sub_expr: str) -> list:
+    """Analyze a sub expression to its operands, by its operator.
+
+    Args:
+        symbol: the operator of the sub expression.
+        sub_expr: the expression to analyze.
+
+    Returns:
+        A list of both operands, left and right.
+    """
+    parsed_expr = sub_expr.split(symbol)
+
+    if symbol == "(":
+        parsed_expr[1] = parsed_expr[1][:-1]  # removes ")"
+
+    return parsed_expr
+
+
+def has_operations(expression: str) -> bool:
+    """Return True if the expression has operations in it else, false."""
+    for symbol in get_operators_by_precedence():
+        if operators[symbol].regex.search(expression):
+            return True
+
+    return False
+
+
+def _evaluate(expression: str) -> float:
+    """Calculate the value of a mathematical expression."""
+    for operator_symbol in get_operators_by_precedence():
+        if not has_operations(expression):
+            return float(expression)
+
+        sub_expression = get_sub_expression_by_symbol(operator_symbol,
+                                                      expression)
+        if not sub_expression:
+            continue
+
+        operands = parse_sub_expression(operator_symbol, sub_expression)
+        result = calculate_sub_expression(operator_symbol, operands)
```

Comment:
if you already make a function to calculate ("evaluate"), why are you using a different one?